pkg/k8s/engine_controller.go

```diff
@@ -149,6 +155,32 @@ func (c *EngineController) shouldHandleGateway(gateway *v1alpha1.Gateway) bool {
 	return false
 }
 
+func (c *EngineController) completeGateway(gateway *v1alpha1.Gateway) error {
+	if gateway.Status.ActiveEndpoint.NodeName != c.nodeName {
+		return nil
+	}
+	publicIP, err := utils.GetPublicIP()
+	if err != nil {
+		return err
+	}
+	// retry to update ip
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
I think we should get the gateway Object from the APIServer on each retry before attempting to update the gateway.

Reply:
fixed